test/object.cpp

```diff
@@ -1469,6 +1502,26 @@ class object_test
         BOOST_TEST(object({{"1",1},{"2",2},{"3",3}}) == object({{"3",3},{"2",2},{"1",1}}));
     }
 
+    void
+    testAllocation()
+    {
+        counting_resource res;
+        {
+            storage_ptr sp(&res);
+            object o(sp);
+            o.reserve(1);
+        }
+        BOOST_TEST(res.allocated == res.deallocated);
```

Comment:
@glenfe can you comment on whether this is a correct way to test that the same amount which was allocated was then deallocated?

Reply:
@grisumbras Yes, it is.